com.visualstudio.code.json

```diff
@@ -0,0 +1,125 @@
+{
+  "app-id": "com.visualstudio.code",
+  "branch": "stable",
+  "base": "io.atom.electron.BaseApp",
+  "base-version": "stable",
+  "runtime": "org.freedesktop.Sdk",
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "code",
+  "tags": ["proprietary"],  
+  "separate-locales": false,
+  "finish-args": [
+      "--share=ipc", "--socket=x11",
+      "--socket=pulseaudio",
+      "--share=network",
+      "--device=dri",
+      "--filesystem=host",
+      "--talk-name=org.freedesktop.Notifications",
+      "--talk-name=org.freedesktop.secrets",
+      "--extra-data=code.deb:57b5f003d7dc13d3555e5b62d7e14a3e9cef7d759e5d2df2432a28d526f38db6:43469528::https://go.microsoft.com/fwlink/?LinkID=760868"
+    ],
+  "modules": [
+    {
+        "name": "libsecret",
+        "config-opts": [
+            "--disable-manpages",
+            "--disable-gtk-doc",
+            "--disable-static",
+            "--disable-introspection"
+        ],
+        "cleanup": [
+            "/bin",
+            "/include",
+            "/lib/pkgconfig",
+            "/share/gtk-doc",
+            "*.la"
+        ],
+        "sources": [
+            {
+                "type": "archive",
+                "url": "http://ftp.gnome.org/pub/GNOME/sources/libsecret/0.18/libsecret-0.18.5.tar.xz",
+                "sha256": "9ce7bd8dd5831f2786c935d82638ac428fa085057cc6780aba0e39375887ccb3"
+            },
+            {
+                "type": "shell",
+                "commands": ["autoreconf -f"]
+            }
+        ]
+    },
+    {
+        "name": "xkbfile",
+        "cleanup": ["/include", "/lib/*.la", "/lib/pkgconfig"],
+        "config-opts": ["--disable-static"],
+        "sources": [
+        {
+            "type": "archive",
+            "url": "https://www.x.org/releases/individual/lib/libxkbfile-1.0.9.tar.bz2",
+            "sha256": "51817e0530961975d9513b773960b4edd275f7d5c72293d5a151ed4f42aeb16a"
+        }
+        ]
+    },
+    {
+        "name": "git",
+        "make-args": ["NO_PERL=1"],
+        "make-install-args": ["NO_PERL=1"],
+        "sources": [
+            {
+                "type": "archive",
+                "url": "https://www.kernel.org/pub/software/scm/git/git-2.14.1.tar.xz",
+                "sha256": "6f724c6d0e9e13114ab35db6f67e1b2c1934b641e89366e6a0e37618231f2cc6"
+            }
+        ]
+    },
+    {
+        "name": "nodejs",
```

Comment:
Probably reasonable to clean up: `/include` `/share` `/lib/node_modules/npm/changelogs` `/lib/node_modules/npm/doc` `/lib/node_modules/npm/html` `/lib/node_modules/npm/man` `/lib/node_modules/npm/scripts`, etc